### ISSUE TITLE
Discrete Updater with SparseCat

### ISSUE THREAD

**Issue by MaximeBouton** [JuliaPOMDP/BeliefUpdaters.jl#5]
It might be a good idea to provide a default implementation of a discrete updater using `SparseCat` since there is usually a significant speed up by using these sparse representations instead of the regular vector one. 
It seems sufficiently useful and problem agnostic to be implemented by default.

**Comment by zsunberg** [comment]
Yeah I think this is a good idea. One thing to keep in mind is that
pdf(::SparseCat, x) is slow (O(n)) - not sure if that will be a problem.

(sent from phone)

**Comment by zsunberg** [comment]
Actually it might be easier to just do this with a sparse vector in a DiscreteBelief

**Comment by MaximeBouton** [comment]
When I was vectorizing `BeliefGridValueIteration`, I found myself writing an update for `SparseTabularPOMDP`: 
```
function update(b::Vector{R}, pomdp::SparseTabularPOMDP, a, o) where R<:Real
    T = transition_matrix(pomdp, a)
    O = view(observation_matrix(pomdp, a), :, o)
    bp = O .* T'* b
    if sum(bp) ≈ zero(R)
        fill!(bp, 1)
    end
    return normalize!(bp, 1)
end
```

It does not match the interface as is but it could eventually make it in this package.
